server/text_generation_server/models/custom_modeling/flash_llama_modeling.py

```diff
@@ -435,3 +462,33 @@ def forward(
             hidden_states = hidden_states[lm_head_indices]
         logits, speculative_logits = self.lm_head(hidden_states)
         return logits, speculative_logits
+
+    # If this function is called, it should always initialize KV cache scale
+    # factors (or else raise an exception). Thus, handled exceptions should
+    # make sure to leave KV cache scale factors in a known good (dummy) state
+    def load_kv_cache_scales(self, quantization_param_path: str) -> None:
+        for layer_idx, scaling_factor in kv_cache_scales_loader(
+            quantization_param_path,
+            self.tp_rank,
+            self.tp_world_size,
+            self.config.num_hidden_layers,
+            self.config.__class__.model_type,
+        ):
+            layer_self_attn = self.model.layers[layer_idx].self_attn
+
+            if SYSTEM == "rocm":
+                # The scaling factor convention we are assuming is
+                # quantized_value * scaling_factor ~= true_value
+                # which is consistent with the practice of setting
+                # scaling_factor = tensor_amax / FPtype_max
```

Comment:
And why does that imply that rocm needs a factor of 2 ?


If the reason is that much global, shouldn't it be handled directly on load?

Reply:
@Narsil I believe it is related to https://onnx.ai/onnx/technical/float8.html and the diff between e4m3fn and e4m3 (different exponent bias). Is that so @mht-sharma?

But shouldn't it be based on this param https://github.com/vllm-project/vllm/blob/319ad7f1d386699e94f629341c9988a926821f24/tests/fp8_kv/llama2-7b-fp8-kv/kv_cache_scales.json#L4 ? Also, we have to kind of decide whether we want to follow the scheme in vllm to store quantization params in a json and keep compatibility for e.g. models on the Hub, or not.

Reply:
Comment from AMD in vllm: 

> We do *2 only for HIP, to deal with the difference in numeric from our chip. after *2 overall effect is identical as without it on NV.

Reply:
Yes, this is the difference between e4m3fn and e4m3 formats. If using a json like the one liked, there should be a check on the `dtype`.